pytentiostat/config_reader.py

```diff
@@ -38,10 +37,126 @@ def parse_config_file(configlocation=None):
+
+    If finds a parameter that is incorrect, will prompt user what is incorrect.
+    Parameters
+    ----------
+    config_data : dict
+        Dictionary containing the data for parameters in the config file.
+
+    Returns
+    -------
+    None
+
+    """
+    #Obtain all the parameters in parsable way
+    average_number = config_data["advanced_parameters"]["average_number"]
+    conversion_factor = config_data["advanced_parameters"]["conversion_factor"]
+    cv_start_voltage = config_data["cyclic_voltammetry"]["start_voltage"]
+    cv_sweep_rate = config_data["cyclic_voltammetry"]["sweep_rate"]
+    cycle_number = config_data["cyclic_voltammetry"]["number_of_cycles"]
+    data_out_name = config_data["general_parameters"]["data_output_filename"]
+    data_out_path = config_data["general_parameters"]["data_output_path"]
+    end_voltage = config_data["linear_sweep_voltammetry"]["end_voltage"]
+    exp_time = config_data["chronoamperometry"]["time"]
+    exp_type = config_data["general_parameters"]["experiment_type"]
+    lsv_start_voltage = config_data["linear_sweep_voltammetry"]["start_voltage"]
+    first_turnover = config_data["cyclic_voltammetry"]["first_turnover_voltage"]
+    lsv_sweep_rate = config_data["linear_sweep_voltammetry"]["sweep_rate"]
+    rest_time = config_data["general_parameters"]["rest_time"]
+    second_turnover = config_data["cyclic_voltammetry"]["second_turnover_voltage"]
+    set_gain = config_data["advanced_parameters"]["setpoint_gain"]
+    set_offset = config_data["advanced_parameters"]["setpoint_offset"]
+    shunt_resistor = config_data["advanced_parameters"]["shunt_resistor"]
+    step_number = config_data["general_parameters"]["step_number"]
+    time_step = config_data["advanced_parameters"]["time_step"]
+    voltage = config_data["chronoamperometry"]["voltage"]
+    
+    #Check if every variable is of the correct type
+    for i in [data_out_name, data_out_path]:
+        bool = isinstance(i, str)
+        if bool == False:
+            srti = str(i)
+            sys.exit("Warning! \nThe value " + stri + " in config.yml is not a string. \nExiting...")
+                      
+    
+    for i in [average_number, cycle_number, step_number]:
+        bool = isinstance(i, int)
+        if bool == False:
+            srti = str(i)
+            sys.exit("Warning! \nThe value " + stri + " in config.yml is not an integer. \nExiting...")
+                
+    for i in [conversion_factor, cv_start_voltage, cv_sweep_rate, end_voltage, 
+              exp_time, first_turnover, lsv_start_voltage, lsv_sweep_rate, rest_time, 
+              second_turnover, set_gain, set_offset, shunt_resistor, time_step,
+              voltage]:
+        bool = isinstance(i, float)
+        if bool == False:
+            bool = isinstance(i, int)
+        if bool == False:
+            srti = str(i)
+            sys.exit("Warning! \nThe value " + stri + " in config.yml is not a number. \nExiting...")
```

Comment:
I think this works fine but it seems a little confusing. Might consider changing to 
`if isinstance(i, float) or isinstance(i, int) == False:
    srti = str(i)
    sys.exit(...)`
to shorten it up and make it more concise.

Reply:
Good suggestions. Incorportaed this + similar for string and int checks too.